packages/plugins/validation-cache/src/index.ts

```diff
@@ -50,18 +50,30 @@ export const useValidationCache = (pluginOptions: ValidationCacheOptions = {}):
     onParse({ params, extendContext }) {
       extendContext({ [rawDocumentSymbol]: params.source.toString() });
     },
-    onValidate({ params, context, setResult }) {
-      const key: string = context[rawDocumentSymbol] ?? print(params.documentAST);
-      const cachedResult = resultCache.get(key);
+    onValidate({ params, context, setValidationFn, validateFn }) {
+      // We use setValidateFn over accessing params.rules directly, as other plugins in the chain might add more rules.
+      // This would cause an issue if we are constructing the cache key here already.
+      setValidationFn((...args) => {
+        let ruleKey = '';
+        if (Array.isArray(args[2])) {
+          // Note: We could also order them... but that might be too much
+          for (const rule of args[2]) {
+            ruleKey = ruleKey + rule.name;
+          }
```

Comment:
I would also prefer to hash the validation names... but we currently have no way of doing cross-platform hashing in a sync and performant way. 😅 We could use something like [emn178/js-sha1@master/src/sha1.js](https://github.com/emn178/js-sha1/blob/master/src/sha1.js?rgh-link-date=2022-12-16T19%3A54%3A56Z) instead for now.

Thoughts?